packages/dcos-net/extra/dcos-net-setup.py

```diff
@@ -70,8 +74,28 @@ def add_networkd_config(src):
         result.returncode = 0
         return result
 
+    # Restart networkd only if the configuration is updated
+    mtime = os.path.getmtime(dst)
+    result = subprocess.run(['systemctl', 'show', '--value',
+                             '--property', 'ActiveEnterTimestamp',
+                             networkd], stdout=subprocess.PIPE)
+    if result.returncode == 0:
+        started = datetime.datetime.strptime(
+            result.stdout.strip().decode(),
+            '%a %Y-%m-%d %H:%M:%S %Z')
+        if started.timestamp() > mtime:
+            return result
+
+    # Restart networkd
     return subprocess.run(['systemctl', 'restart', networkd])
 
 
+def safe_filecmp(src, dst):
+    try:
+        return filecmp.cmp(src, dst)
+    except:
```

Comment:
Can we please try to avoid catching all the exceptions and explicitly list the ones that we expect to ignore? This shold make the code more robust.

Reply:
That's a good recommendation in general, in this case I'd really like to copy `src` file to `dst` if any error with `filecmp.cmp` happens. It could be posix error (`enoent`, `eaccess`, etc) or logical error in `filecmp` module, that doesn't matter. `filecmd` must always return boolean value with no exceptions. It catches all errors for one small function call, there is no other logic here. How would it make the code more robust if it catches all possible cases explicitly in this particular case?

Reply:
If you get the permission error, then it does not really make sense to try to copy the file I believe. What is more - the result of the comparison is invalid (you  were not able to compare these files in the end). 

I would rather catch the following errors:
* file does not exist